doc/zmq_unbind.txt

```diff
@@ -14,68 +14,8 @@ int zmq_unbind (void '*socket', const char '*endpoint');
 
 DESCRIPTION
 -----------
-The _zmq_unbind()_ function shall unbind a socket specified
```

Comment:
Why the removal of the unbind documentation?

Reply:
while the underlying implementation might use common code, the behaviour is different, and most importantly they are semantically different - connect/disconnect and bind/unbind is a simple model to understand, while mixing them I'm afraid would just confuse users

Reply:
in other words, the behaviour of a bound socket on unbind and a connected socket on disconnect is quite different, so it makes sense to have differen functions for them

Reply:
> connect/disconnect and bind/unbind is a simple model to understand

Yes it is simpler to understand but thats not how the library works. I agree that a disconnect function whose behaviour change according to the context might confuse users but thats because the function itself is confusing. The underlying implementation is [literally the same](https://github.com/zeromq/libzmq/blob/master/src/zmq.cpp#L316). The current documentation makes it seem like calling disconnect on a bound socket is undefined behaviour and vice verse. As a user I did not expect that at all, and it confused me more than a `disconnect` method whose behaviour depends on the context.

Personally I think that if someone wanted to make disconnect and unbind separate it should first check if the endpoint is indeed connected or bound first and return an error otherwise. However, it is not possible without breaking the API, so new methods should be implemented. I think this would be ideal.

Reply:
In other words, this kinda feels like lying to users to make the API easier to use. I think we should instead make another API with an actual separate behaviour for disconnect vs. unbind, as opposed to pretending like its the case.